cmd/dryrun/dryrun.go

```diff
@@ -0,0 +1,537 @@
+		return nil, err
+	}
+
+	dynamicClient := dynfake.NewSimpleDynamicClient(scheme.Scheme)
+	clientset := clientsetfake.NewSimpleClientset()
+	watcherReconciler, _ := depclient.NewControllerRuntimeSource()
+
+	dynamicWatcher := depclient.NewWithClients(
+		dynamicClient,
+		clientset.Discovery(),
+		watcherReconciler,
+		&depclient.Options{DisableInitialReconcile: true, EnableCache: true},
+	)
+
+	go func() {
+		err := dynamicWatcher.Start(ctx)
+		if err != nil {
+			panic(err)
+		}
+	}()
+
+	runtimeClient := clientfake.NewClientBuilder().
+		WithScheme(scheme.Scheme).
+		WithObjects(configPolCRD, cfgPolicy).
+		WithStatusSubresource(cfgPolicy).
+		Build()
+
+	nsSelUpdatesChan := make(chan event.GenericEvent, 20)
+	nsSelReconciler := common.NewNamespaceSelectorReconciler(runtimeClient, nsSelUpdatesChan)
+
+	serverVersion := ""
+
+	versionInfo, err := clientset.Discovery().ServerVersion()
+	if err == nil {
+		serverVersion = versionInfo.String()
+	}
+
+	rec := ctrl.ConfigurationPolicyReconciler{
+		Client:                 runtimeClient,
+		DecryptionConcurrency:  1,
+		DynamicWatcher:         dynamicWatcher,
+		Scheme:                 scheme.Scheme,
+		Recorder:               record.NewFakeRecorder(8),
+		InstanceName:           "policy-cli",
+		TargetK8sClient:        clientset,
+		TargetK8sDynamicClient: dynamicClient,
+		SelectorReconciler:     &nsSelReconciler,
+		EnableMetrics:          false,
+		UninstallMode:          false,
+		ServerVersion:          serverVersion,
+		EvalBackoffSeconds:     5,
+	}
+
+	if d.mappingsPath != "" {
+		reader, err := os.Open(d.mappingsPath)
+		if err != nil {
+			return nil, err
+		}
+
+		apiMappings := []mappings.APIMapping{}
+		if err := json.NewDecoder(reader).Decode(&apiMappings); err != nil {
+			return nil, err
+		}
+
+		rec.RestMapper = mappings.MapperFrom(apiMappings)
+	} else {
+		mapper, err := mappings.DefaultMapper()
+		if err != nil {
+			return nil, err
+		}
+
+		rec.RestMapper = mapper
+	}
+
+	// wait for dynamic watcher to have started
+	<-rec.DynamicWatcher.Started()
+
+	return &rec, nil
+}
+
+func (d *DryRunner) saveStatus(status policyv1.ConfigurationPolicyStatus) error {
+	f, err := os.Create(d.statusPath)
+	if err != nil {
+		return err
+	}
+
+	out, err := k8syaml.Marshal(status)
+	if err != nil {
+		return err
+	}
+
+	fmt.Fprint(f, out)
+
+	return nil
+}
+
+func (d *DryRunner) saveOrPrintComplianceMessages(
+	ctx context.Context, cmd *cobra.Command, rec client.Client, ns string,
+) error {
+	events := corev1.EventList{}
+
+	if err := rec.List(ctx, &events, client.InNamespace(ns)); err != nil {
+		return err
+	}
+
+	messages := []string{}
+
+	for _, ev := range events.Items {
+		if ev.InvolvedObject.Name == parentName {
+			messages = append(messages, ev.Message)
+		}
+	}
+
+	if d.messagesPath != "" {
+		f, err := os.Create(d.messagesPath)
+		if err != nil {
+			return err
+		}
+
+		for _, msg := range messages {
```

Comment:
Is this already sorted by timestamp?

Reply:
I believe the items in the event list are sorted lexically, but since a timestamp is in the name, effectively it is sorted by time.